mlir/include/mlir/Transforms/DialectConversion.h

```diff
@@ -1241,16 +1241,16 @@ struct ConversionConfig {
   /// 2. Pattern produces IR (in-place modification or new IR) that is illegal
   ///    and cannot be legalized by subsequent foldings / pattern applications.
   ///
-  /// If set to "false", the conversion driver will produce an LLVM fatal error
-  /// instead of rolling back IR modifications. Moreover, in case of a failed
-  /// conversion, the original IR is not restored. The resulting IR may be a
-  /// mix of original and rewritten IR. (Same as a failed greedy pattern
-  /// rewrite.)
+  /// Experimental: If set to "false", the conversion driver will produce an
+  /// LLVM fatal error instead of rolling back IR modifications. Moreover, in
+  /// case of a failed conversion, the original IR is not restored. The
+  /// resulting IR may be a mix of original and rewritten IR. (Same as a failed
+  /// greedy pattern rewrite.) Use MLIR_ENABLE_EXPENSIVE_PATTERN_API_CHECKS
+  /// with ASAN to detect invalid pattern API usage.
```

Comment:
```suggestion
  /// greedy pattern rewrite.) Use the cmake build option
  /// `-DMLIR_ENABLE_EXPENSIVE_PATTERN_API_CHECKS=ON`
  /// (possibly alongside with ASAN) to detect invalid pattern API usage.
```

Reply:
Folks may think that it could be an environment variable or something like that.